### ISSUE TITLE
Dotnet Core 3.1 Not Supported On Azure.

### ISSUE THREAD

**Issue by capraynor** [microsoft/Oryx#473]
Hi,
When will you deploy the latest version to azure app service? 

Now the Dotnet Core 3.1 is not supported on azure app service linux, and it shows that using an old version of Oryx. 

App service Name: 
sjs-...e-test-net-core
Log attached.
```
Command: oryx build /home/site/repository -o /home/site/wwwroot --platform dotnet --platform-version 3.1 -i /tmp/8d7a897238a60cc --log-file /tmp/build-debug.log 
Build orchestrated by Microsoft Oryx, https://github.com/Microsoft/Oryx
You can report issues at https://github.com/Microsoft/Oryx/issues

Oryx Version      : 0.2.20191105.2, Commit: 67e159d71419415435cb5d10c05a0f0758ee8809, ReleaseTagName: 20191105.2
Build Operation ID: |rxQnzkoXvEU=.44a31120_
Repository Commit : 56862e69c90acde5d91714a03eb6331f62f0bef6

Error: Platform 'dotnet' version '3.1' is unsupported. Supported versions: 1.0.16, 1.1.14, 2.0.9, 2.1.13, 2.2.7, 3.0.0
Error: Platform 'dotnet' version '3.1' is unsupported. Supported versions: 1.0.16, 1.1.14, 2.0.9, 2.1.13, 2.2.7, 3.0.0\n/opt/Kudu/KuduConsole/Scripts/starter.sh oryx build /home/site/repository -o /home/site/wwwroot --platform dotnet --platform-version 3.1 -i /tmp/8d7a897238a60cc --log-file /tmp/build-debug.log 
```

Thanks for your concern. 

Raynor

**Comment by ddisqq** [comment]
Also seeing this issue.

**Comment by gpcastro** [comment]
Thanks for letting us know. A new version is rolling out in the next couple of weeks, so you should see the 3.1 SDK soon. @sanchitmehta can give us more details about the roll out.

**Comment by svenskmand** [comment]
Hi :) Any updates on this? :)

**Comment by tryan702** [comment]
I tried an azure linux app service 4 times this evening.  Tried in East US, Central US, West US, and UK South regions and receive the same error:
```
Error: Platform 'dotnet' version '3.1' is unsupported. Supported versions: 1.0.16, 1.1.14, 2.0.9, 2.1.13, 2.2.7, 3.0.0
Error: Platform 'dotnet' version '3.1' is unsupported. Supported versions: 1.0.16, 1.1.14, 2.0.9, 2.1.13, 2.2.7, 3.0.0\n/opt/Kudu/Scripts/starter.sh oryx build /home/site/repository -o /home/site/wwwroot --platform dotnet --platform-version 3.1 -i /tmp/8d7bb21bd5f9c44 --log-file /tmp/build-debug.log 
```
Any updates?  Timing when this will be fixed?

**Comment by sanchitmehta** [comment]
@svenskmand  @tryan702 We have started the rollout for the latest platform release for App Service, you should be able to see ASP .NETCore 3.1 over next two weeks, as we upgrade globally.

**Comment by milisievert** [comment]
Any updates?

**Comment by R-penguins** [comment]
I'm having the same issue. I can choose dotnet 3.1 during deployment but actually orxy doesn't seem to support it.

**Comment by sanchitmehta** [comment]
@milisievert @R-penguins we would complete our upgrades in a few more days, then you should be able to access the ASP .NETCore 3.1. What regions are your apps hosted in ?

**Comment by milisievert** [comment]
@sanchitmehta uk west

**Comment by R-penguins** [comment]
> @milisievert @R-penguins we would complete our upgrades in a few more days, then you should be able to access the ASP .NETCore 3.1. What regions are your apps hosted in ?

East US

**Comment by pkwieci1** [comment]
West EU here.
Seriously, it was my first attempt to publish a web app in my entire life and I was struggling hard to get everything to work (first experiences and troubles with Git etc.). Having insufficient knowledge about how everything works, only a huge amount of luck in finding the error and then this thread has prevented me from having a nervous breakdown, lol.

**Comment by mijabr** [comment]
works on AWS, haha

**Comment by LukeStephen** [comment]
Same, need this for my East US 2

**Comment by capraynor** [comment]
@sanchitmehta 
> We have started the rollout for the latest platform release for App Service, you should be able to see ASP .NETCore 3.1 over next two weeks, as we upgrade globally

MS team said like this few month ago, and now 4 weeks passed since your post. 

What's going on with this issue then?

**Comment by pkwieci1** [comment]
I think I found a solution.
Look into App Services -> Configuration -> General Settings -> Major/Minor Version and set it to 3.1.
I could finally deploy my app to the cloud. The fact I'm getting an ":( Application Error" is another thing but that may be my fault.

**Comment by milisievert** [comment]
@pkwieci1 mine were set to 3.1 . Still not supported ...

**Comment by tambu22** [comment]
I have a work around, I'm building in a amazon spot instance and deploying the zip file via Azure Cli

**Comment by R-penguins** [comment]
My build finally seems to be working! 🎉 The orxy version has changed to 0.2.20200114.13 on Azure hosting in East US. Thanks!

**Comment by R-penguins** [comment]
By the way, could anyone kindly point to me where to find help on this issue?

```
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'System.Text.Json, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
File name: 'System.Text.Json, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
    at Microsoft.Extensions.DependencyInjection.JsonProtocolDependencyInjectionExtensions.AddJsonProtocol[TBuilder](TBuilder builder, Action`1 configure)
```
My solution works fine locally on IIS, but failed after deployment on Azure..

**Comment by capraynor** [comment]
@R-penguins
Hi, 

have you ever tried to "Copy referenced assemblies to local" when build? 

Have you checked that the assembly "System.Text.Json" is in artifact or not?

**Comment by R-penguins** [comment]
> @R-penguins
> Hi,
> 
> have you ever tried to "Copy referenced assemblies to local" when build?
> 
> Have you checked that the assembly "System.Text.Json" is in artifact or not?

Thanks for pointing out! I added the "System.Text.Json" NuGet package (which I thought was included by default in dotnet 3.1...) and it worked😂 so excited 😆

**Comment by kichalla** [comment]
AppService has finished deploying this, so it should be available to you.
